src/controllers/forgotPassword.js

```diff
@@ -0,0 +1,80 @@
+import jwt from 'jsonwebtoken';
+import nodemailer from 'nodemailer';
+import { User } from '../database/models';
+import checkEmailValidation from '../validations/reset-checkEmail';
+import template from '../resetEmail';
+
+/**
+ * Set a controller Class resetPassword
+ */
+// class forgotPasswordController {
+module.exports = {
+  // Check if the email is provided
+  async forgot(req, res) {
+    if (!req.body.email) {
+      res.json({ error: 'Please insert your email !!!' });
+    }
+
+    // Check email validation
+    const { error } = await checkEmailValidation(req.body);
+    if (error) {
+      return res.status(400).json({ err: error.details[0].message });
+    }
+
+    // Check if the email exists
+    const emailExist = await User.findOne({
+      where: { email: req.body.email }
+    });
+    if (!emailExist) {
+      return res
+        .status(409)
+        .json({ message: res.__('Email does NOT exist in our database !!!') });
+    }
+
+    // Save seconds from 1st January 1970
+    const token = jwt.sign({ id: User.id }, process.env.RESET_PASSWORD_KEY, {
+      expiresIn: '1440m'
+    });
```

Comment:
Don't use jwt like this, the are helper functions that handle this. Check this [file](https://github.com/atlp-rwanda/script-heroes-bn-backend/blob/develop/src/utils/jwtFunctions.js)

Reply:
Now encode & decode are used.